### PR TITLE
Release 128tech 1.22 cherry pick 128tech 1.17

### PR DESCRIPTION
**Pull request by gregschrock**
# Update to Upstream 1.22 Branch

## New Work

One commit of infrastructure updates

## Testing

- [X] Unit tests pass
- [X] Monitoring agent commands robot test passes (needed to change `${AGENT_EXTRA_LOG_LINES}` to `12`)
- [X] Monitoring agent plugin robot test passes (Kafka install was failing for unrelated reasons, but Kafka works based on the MA commands robot test)

## Cherry Pick Contents

Cherry picked changes from the 1.17 branch. Some common issues that needed to be resolved.

- `internal.Duration` was moved to `config.Duration` which is now just an alias for `time.Duration`
- the metrics library `metrics.New` no longer returns an error as a second return value

### PR (3): Unix sockets for http output plugin

https://github.com/128technology/telegraf/pull/3
- 0e6cf7400: Unix sockets for http output plugin

### PR (2): Use docker for consistent RPM build environment

https://github.com/128technology/telegraf/pull/2
- bfe626aba: Create a docker-env script to simplify the building of telegraf-128tech RPMs

### PR (6): cherry-pick execd fix for one metric per batch

https://github.com/128technology/telegraf/pull/6
- 0c920ecbf: Add some more useful information about maintaining the 128tech branching model

### PR (7): Create the t128_metrics input plugin

https://github.com/128technology/telegraf/pull/7
- aa778a1a7: Create the t128_metrics input plugin
- 0308c26b0: Address review comments

### PR (8): Add CI to 128tech branches

https://github.com/128technology/telegraf/pull/8
- 5f8eb62db: Add CI to 128tech telegraf fork

### PR (9): Add a release workflow for simpler RPM building on release

https://github.com/128technology/telegraf/pull/9
- ad090e4b6: Add a release workflow for simpler RPM building on release

### PR (11): Fix t128_metrics timeout parsing error

https://github.com/128technology/telegraf/pull/11
- 8b9af04b6: Convert t128_metrics timeout to internal.Duration to be correctly parsed
- e2ab997ed: Update t128_metrics_test

### PR (12): Account for patch in version tag

https://github.com/128technology/telegraf/pull/12
- 5bd9aa0b0: Add a script that more intelligently parses the version tag

### PR (16): Cherry pick 128tech 1.14 changes to 1.16

https://github.com/128technology/telegraf/pull/16
- 7d23e176d: Reference patch input in action.yaml

### PR (13): Include Content-Type in t128_metrics HTTP requests

https://github.com/128technology/telegraf/pull/13
- 7c1f8dd8e: Include content-type in t128_metrics HTTP requests

### PR (14): Update t128_metrics integer conversion to be optional

https://github.com/128technology/telegraf/pull/14
- 518a76c03: Update t128_metrics integer conversion to be optional

### PR (25): Implement GraphQL Collector (really just setup for GraphQL Collector)

https://github.com/128technology/telegraf/pull/25
- 898bff1af: Add influx_timestamp_units to the influx output serializer
- 9c9bce9b4: Correct indicated default units
- 70d674cb4: Allow the t128_metrics input to retrieve metrics in bulk
- 58df9faf3: Implement the 128t_rate processor
- 2673acc5e: Update t128_rate to t128_transform

### PR (25): Implement GraphQL Collector (build updates)

https://github.com/128technology/telegraf/pull/25
- 006c0f26d: Update some documentation/CI based on telegraf 1.16 changes
- f59c12d06: Update assignment of environment variables to avoid security vulnerability
- 73de0cb97: Update github actions to automatically publish (to github release page) the built RPM
- 7a3e14936: Update with new upstream build approach
- 96bebb2fa: Fix docker env build

### PR (21): Cherry pick changes from 128tech 1.16 to 128tech 1.17

https://github.com/128technology/telegraf/pull/21
- 49a01b9cd: Update github actions to tage tags within github'"'"'s restrictions

### PR (27): Implement GraphQL Collector

https://github.com/128technology/telegraf/pull/27
- 8fbd0814a: Base branch for graph collector
- 38062754a: Addressed all minor PR comments
- 6b41fa7b0: Addressed PR comments related to unit tests
- e4b67d613: More PR nits
- 80f8f9458: Support nested tags and fields for deviceState
- 46e6c11a7: Clean code, add docstrings and comments
- 4c525e9e7: Support nested tags and fields for deviceState
- 8e9fb67bc: Clean code, add docstrings and comments
- f0b540f84: Address PR comments
- 2048123d7: Add support for complex predicates
- a4c4fd10f: Working for mist agent queries
- d7de03e67: Code cleanup
- aabce79af: Cleanup and more tests

### PR (28): Allow tagless t128graphql config

https://github.com/128technology/telegraf/pull/28
- b3e7a2b0c: Allow tagless t128graphql config

### PR (30): Sort the request IDs during construction for consistent ID ordering

https://github.com/128technology/telegraf/pull/30
- 9345562eb: Sort the request IDs during constuction for consistent ID ordering

### PR (29): Add graphql input readme

https://github.com/128technology/telegraf/pull/29
- d54c32143: add graphql readme
- 4554f3e8a: Add more detail to readme
- d89bd9b24: Ouput and aggregator for Zabbix
- 946de3e02: Revert "Ouput and aggregator for Zabbix"

### PR (32): Update sarama library to latest in 128tech fork and added batch retry mechanism

https://github.com/128technology/telegraf/pull/32
- 1ccbd63cb: Upgrade sarama to 1.29.0
- 4749c8df7: Updated sarama go.mod entry with 128tech repo
- 6c93c455d: Log each error from kafka
- 54ec79aa8: Added index and message as string
- f4e81be79: Reworked message sending logic to retry 5 times with the messages that failed
- 78ef49e4f: Create copy of message so that it can'"'"'t be garbage collected
- dceb0fcbc: Added max_batch_retry driven by config and fixed formatting of message
- db91f517d: Added MaxBatchRetry to all kafka tests
- e48cff548: Addressed comments and added unit testing
- dbefe9811: Removed left over prints

### PR (33): Create a state-change transform type for the t128_transform processor

https://github.com/128technology/telegraf/pull/33
- ef1ed8617: Create a state-change transform type for the t128_transform processor
- 49afcd282: Add the ability to report previous values in the t128_transform processork
- aa26bb7d4: Selectively use the transformed value as the previous one

### PR (34): Support absolute path to extract tags/fields in graph collector

https://github.com/128technology/telegraf/pull/34
- eeb93f36c: support other tags in graphql collector
- 048aabcff: move other_tags under extract_fields and extract_tags
- 9c3065de0: rename some vars and funcs

### PR (35): Added start of filter processor

https://github.com/128technology/telegraf/pull/35
- 0b01ff2bb: Added start of filter processor
- 999599123: Updated implementation to match revised functional spec
- be22e854d: Refactor code to be in terms of matchers
- af422e524: Removed extra quotes
- 00e98708d: Addressed review comments

### PR (36): Supported regex and glob in filter processor

https://github.com/128technology/telegraf/pull/36
- b7ef5fade: Supported regex and glob in filter processor
- d0d462dde: Updated sample config and readme and added invalid mode test
- 382df1034: Addressed review comments

### PR (37): Added operation config choice for each condition

https://github.com/128technology/telegraf/pull/37
- 34e2aaa5d: Added operation config choice for each condition
- 7cc83ea15: Fixed indentation
- cd29cb9ba: Fixed plurality
- ce130c7fd: Added to toml loading testcase

### PR (38): Support inversion

https://github.com/128technology/telegraf/pull/38
- da5b9a3ee: Added invert flag to invert whole condition

### PR (41): Pass processor

https://github.com/128technology/telegraf/pull/41
- 4863386af: Correctly drop the metrics
- 1069ca085: Renamed processor from filter to pass

### PR (42): Support field filtering

https://github.com/128technology/telegraf/pull/42
- 57ee03d18: Added field filtering support to t128_pass
- a8d51cfa9: Added back top level readme changes
- 70a7a9d9d: Fixed bug for non-string values and added test
- d1d66a6a2: Enhanced tests
- 03e043545: Added support for ignore_missing_keys option
- f83d11394: Added string section below fields to allow other types later
- aeb9b2ccc: Fixed another reference in the readme

### PR (43): Escaped invalid syslog characters for structured data parameters

https://github.com/128technology/telegraf/pull/43
- 83071c554: Escaped invalid syslog characters for structured data parameters

### PR (44): Persist t128 transform state change cache

https://github.com/128technology/telegraf/pull/44
- 2960692e2: add t128 transform cache
- fbab4081a: review feedback
- 2dfbfb420: update readme
- 62b2e8efb: review feedback
- 0e5479f21: add missing check

### PR (45): Dont retry metric retrieval if configured endpoint is not found

https://github.com/128technology/telegraf/pull/45
- 2d98a8fcb: dont retry metrics if no configured enpoint is found
- f9d60fe88: gofmt
- 5df909480: add t128_metrics unit test
- 739cceccf: add t128_graphql retry config option
- 1a8c7e4c0: tweaks from manual test
- f8a47aa97: gofmt
- 97fadd606: nits

### PR (46): Add an encoding processor for encoding/decoding gzip in string fields

https://github.com/128technology/telegraf/pull/46
- 23fcfedb2: Add an encoding processor for encoding/decoding gzip in string fields

### PR (47): Encoding processor handle standard or raw base64 encoding

https://github.com/128technology/telegraf/pull/47
- 72dc3de09: Encoding processor handle standard or raw base64 encoding

### PR (48): Handle errors in graph collector and support deadline header

https://github.com/128technology/telegraf/pull/48
- 02d125539: handle errors in t128_graphql
- 6233752f8: cleanup
- 4b28fda99: undo changes
- f616f5bed: add newline
- a15fca5e4: fix test
- 4351e6561: add deadline header
- 0b714bde8: default deadline is 0, more testing
- 2c7e8784e: review feedback

### PR (49): Graph collector deadline header v2

https://github.com/128technology/telegraf/pull/49
- 7d1d48696: implicitly calculate deadline

### CODE REVIEW

File: plugins/outputs/http/http.go.orig [hunk 1]
```diff
@@ -0,0 +1,371 @@
+package http
```
Comment (agrawalkaushik) [inline plugins/outputs/http/http.go.orig]: remove?

Comment (gregschrock) [reply plugins/outputs/http/http.go.orig]: I rewrote history to avoid the one separate file removal commit 🤷🏼‍♂️

**Review by salegria8 — APPROVED** [review]
what's the reasoning behind the change again? There's some feature in 1.22 that we need but I forget what it is

**Review by BenMatase — APPROVED** [review]

**Comment by gregschrock** [comment]
> what's the reasoning behind the change again? There's some feature in 1.22 that we need but I forget what it is

The particular improvement we wanted was https://github.com/influxdata/telegraf/pull/9488 which adds cgroup globs and systemd children.

**Pull request merged**